courses/models.py

```diff
@@ -1117,6 +1119,34 @@ def update_mode_and_save(self, mode, no_user=False):
         self.enrollment_mode = mode
         return self.save_and_log(None if no_user else self.user)
 
+    @transaction.atomic
+    def save_and_log(self, acting_user, *args, **kwargs):
+        """
+        Saves the object and creates an audit object.
+
+        Args:
+            acting_user (User):
+                The user who made the change to the model. May be None if inapplicable.
+        """
+        before_obj = self.objects_for_audit().filter(id=self.id).first()
+        self.save(*args, **kwargs)
+        self.refresh_from_db()
+        before_dict = None
+        if before_obj is not None:
+            before_dict = before_obj.to_dict()
+
+        call_stack = "".join(traceback.format_stack()[-6:-2])
+
+        audit_kwargs = dict(
+            acting_user=acting_user,
+            modified_by=call_stack,
```

Comment:
Not sure if any side effect but why can't it be handled in https://github.com/mitodl/mitxonline/blob/main/main/models.py#L72 since we are adding `modified_by` to AuditModel base model anyway?